File: pyround1/326.3-的幂.py
#
# @lc app=leetcode.cn id=326 lang=python
#
# [326] 3的幂
#

# @lc code=start
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True
        while(n > 3):
            if (n%3 == 0):
                n //= 3
            else:
                break
        return n == 3
        
# @lc code=end

File: 326.3-的幂.cpp
/*
 * @lc app=leetcode.cn id=326 lang=cpp
 *
 * [326] 3的幂
 */

// @lc code=start
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n == 0)
            return false;
        while(n > 1){
            if(n % 3 != 0)
                return false;
            n /= 3;
        }
        return n == 1;
    }
};
// @lc code=end

